inst/include/Rcpp/platform/compiler.h

```diff
@@ -56,6 +43,13 @@
     #if GCC_VERSION >= 40600
         #define IS_GCC_460_OR_LATER
     #endif
+    // Inform users of dated compiler (e.g. <= RHEL6 )
```

Comment:
Per my other comment, I don't think we can do this. We have 700+ packages on CRAN all of whom will be subjected to this warning (and all of whom will have to field enquiries about the warning from users and sysadmins worried about it's implications). If we believe that GCC <= 4.6.2 truly warrants a warning message then we need to *change our code* so it doesn't.